README.md

```diff
@@ -330,6 +331,65 @@ end
 
 5. Your app is now able to access the authenticated `ProductsController` from the `HomeController` using session tokens.
 
+### Marking shops as uninstalled on app/uninstalled webhook
+
+To ensure OAuth still works with session tokens, your app must update its shop records in the event a shop uninstalls your app. You can be notified of these events by subscribing to the `app/uninstall` [webhook](https://shopify.dev/docs/admin-api/rest/reference/events/webhook).
+
+#### Setup app/uninstall webhook
+1. Use the `add_webhook` generator from ShopifyApp to setup the `app/uninstalled` webhook for the app.
+
+```shell script
+rails g shopify_app:add_webhook -t app/uninstalled -a {your_app_url}/webhooks/app_uninstalled
+```
+
+This will add the `app/uninstalled` webhook to your config and create the `AppUninstalledJob` job class for you.  
+
+2. Update the `AppUninstalledJob` job class to mark the Shop record as uninstalled.
+
+```ruby
+class AppUninstalledJob < ActiveJob::Base
+  def perform(args)
+    shop = Shop.find_by(shopify_domain: args[:shop_domain])
+
+    mark_shop_as_uninstalled(shop)
+  end
+
+  private
+
+  def mark_shop_as_uninstalled(shop)
+    shop.destroy! if shop
+  end
+end
+```
+
+In this example, the app marks the shop as uninstalled by deleting the Shop record.
+
+3. Define a background job to ensure existing shops also have the uninstall webhook setup.  
+In this app, the `RegisterWebhooksForActiveShop` job is defined to iterate all shop records in the database and configure the webhooks.
```

Comment:
Should we add a note that this is only needed for apps that have already been installed on shops?

Reply:
```suggestion
3. Define a background job to ensure shops with existing installations of the app also have the uninstall webhook set up.  
In this app, the `RegisterWebhooksForActiveShop` job is defined to iterate all shop records in the database and configure the webhooks.
```